tests/unit/CoreQuoteTest.php

```diff
@@ -0,0 +1,328 @@
+
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+
+		$this->assertNotEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 1, count( $block['innerBlocks'] ), 'There should be only one inner block' );
+		$this->assertEquals( 'core/paragraph', $block['innerBlocks'][0]['name'], 'The inner block name should be core/paragraph' );
+
+		$this->assertEquals( 'core/quote', $block['name'], 'The block name should be core/quote' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		unset( $block['attributes']['citation'] ); // Tested above.
+		unset( $block['attributes']['className'] ); // Tested above.
+		unset( $block['attributes']['cssClassName'] ); // Tested above.
+
+		// Verify the attributes.
+		$this->assertEquals(
+			[
+				'anchor'          => 'test-anchor', // Previously untested.
+				'backgroundColor' => 'pale-cyan-blue', // Previously untested.
+				'fontFamily'      => 'body', // Previously untested.
+				'fontSize'        => 'small', // Previously untested.
+				'gradient'        => 'pale-ocean', // Previously untested.
+				'lock'            => wp_json_encode( // Previously untested.
+					[
+						'move'   => true,
+						'remove' => true,
+					]
+				),
+				'style'           => wp_json_encode( // Previously untested.
+					[
+						'elements' => [
+							'heading' => [
+								'color' => [
+									'text'       => 'var:preset|color|vivid-cyan-blue',
+									'background' => 'var:preset|color|cyan-bluish-gray',
+								],
+							],
+						],
+					]
+				),
+				'textColor'       => 'vivid-red', // Previously untested.
+				'value'           => '<p>Sample Quote</p>', // Previously untested.
+			],
+			$block['attributes'],
+		);
+	}
+
+
+	/**
+	 * Test case for retrieving core quote block untested attributes.
+	 *
+	 * Covers : 'layout' and 'textAlign'.
+	 */
+	public function test_retrieve_core_quote_errorneous_attributes(): void {
+		// layout and textAlign are only supported in WP 6.5+.
+		if ( ! is_wp_version_compatible( '6.5' ) ) {
+			$this->markTestSkipped( 'This test requires WP 6.6 or higher.' );
+		}
+		$block_content = '
+			<!-- wp:quote {"layout":"test-layout","textAlign":"center"} -->
+			<blockquote class="wp-block-quote" id="test-anchor"><!-- wp:paragraph -->
+			<p>Sample Quote</p>
+			<!-- /wp:paragraph --><cite>Citation</cite></blockquote>
+			<!-- /wp:quote -->
+		';
+
+		// Update the post content with the block content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
```

Comment:
You're still using the shared query instead of the 6.6-specific attributes. See my latest commit on CoreList for an example of a streamlined fragment.